src/utils.c

```diff
@@ -161,6 +161,28 @@ SEXP vctrs_set_attributes(SEXP x, SEXP attrib) {
   return x;
 }
 
+SEXP new_list_of(SEXP x, SEXP ptype) {
```

Comment:
Maybe in `type-list-of.c`?

Reply:
We should probably move the data frame utils to `type-dataframe.c` eventually.

Reply:
✅ Maybe in `type-list-of.c`?

> move the data frame utils to type-dataframe.c eventually.

Hasn't that been done already? https://github.com/r-lib/vctrs/blob/8d12bfc0e29e056966e0549af619253253752a64/src/type-data-frame.c#L6